lib/services/files-hash-service.ts

```diff
@@ -0,0 +1,26 @@
+import { executeActionByChunks } from "../common/helpers";
+import { DEFAULT_CHUNK_SIZE } from "../common/constants";
+
+export class FilesHashService implements IFilesHashService {
+	constructor(private $fs: IFileSystem) { }
+
+	public async generateHashes(files: string[]): Promise<IStringDictionary> {
+		const result: IStringDictionary = {};
+
+		const action = async (file: string) => {
+			if (this.$fs.getFsStats(file).isFile()) {
```

Comment:
maybe you should try-catch this call, in case someone deletes the file while we are executing a previous chunk